posthog/temporal/data_modeling/run_workflow.py

```diff
@@ -577,6 +582,33 @@ async def hogql_table(query: str, team: Team, logger: FilteringBoundLogger):
             yield table_from_py_list(batch, pa_schema)
 
 
+def _transform_unsupported_decimals(batch: pa.Table, logger: FilteringBoundLogger) -> pa.Table:
+    """
+    Transform high-precision decimal columns to types supported by Delta Lake.
+    Delta Lake supports decimal types up to precision 38, but ClickHouse can return
+    Decimal256 types with 76 digit precision.
+    """
+    schema = batch.schema
+    columns_to_cast = {}
+
+    for field in schema:
+        if pa.types.is_decimal(field.type):
+            decimal_type = field.type
+            if hasattr(decimal_type, "precision") and decimal_type.precision > 38:
+                columns_to_cast[field.name] = pa.float64()
```

Comment:
Hmm, I dont think we wanna change the underlying type here. Using a float loses precision (especially in large scale numbers). Just purely downscailing to 38 precsion decimal 128 is likely what we want instead - we retain the underlying decimal type and it's more precise for these longer numbers.

Reply:
Ah, ok this I was curious about this one... I tried to do this one first but the type coercion wasn't as clean as I expected, but I'll revisit the approach.

Reply:
I'm not totally sure what the best approach is if the decimal128 is not enough for the input. My anecdotal (from looking at a couple of our logs) understanding is that most of the digits that we see do not break this bound, but it is possible, there is a test that has this case.